docarray/array/mixins/attribute.py

```diff
@@ -1,25 +1,42 @@
-from typing import List, Union
+from typing import TYPE_CHECKING, List, Union
 
 from docarray.array.abstract_array import AbstractDocumentArray
 from docarray.document import BaseDocument
 
+if TYPE_CHECKING:
+    from docarray.typing import NdArray, TorchTensor
+
 
 class GetAttributeArrayMixin(AbstractDocumentArray):
     """Helpers that provide attributes getter in bulk"""
 
-    def _get_documents_attribute(
-        self, field: str
-    ) -> Union[List, AbstractDocumentArray]:
+    def _get_array_attribute(
+        self,
+        field: str,
+    ) -> Union[List, AbstractDocumentArray, 'TorchTensor', 'NdArray']:
         """Return all values of the fields from all docs this array contains
 
         :param field: name of the fields to extract
         :return: Returns a list of the field value for each document
         in the array like container
         """
-
         field_type = self.__class__.document_type._get_nested_document_class(field)
 
-        if issubclass(field_type, BaseDocument):
+        if (
+            self.is_stacked()
+            and field in self._column_fields()
+            and self._columns is not None
+        ):
+            attributes = self._columns[field]
+            if attributes is not None:
+                return attributes
+            else:
+                raise ValueError(
+                    f'The column is not set for the field {field} even though '
+                    f'it is in stacked mode'
```

Comment:
I think "column" is not a term known to the user in this context
```suggestion
                    f'The DocumentArray is in stacked mode, but no stacked data is present.
```
When can this even happen? What is the user supposed to do about it? can we communicate something that helps to resolve the issue?